bokehjs/src/coffee/document.coffee

```diff
@@ -650,8 +651,13 @@ export class Document
             throw new Error("Cannot apply patch to #{patched_id} which is not in the document")
           patched_obj = @_all_models[patched_id]
           attr = event_json['attr']
-          value = Document._resolve_refs(event_json['new'], old_references, new_references)
-          patched_obj.setv({ "#{attr}" : value })
+          model_type = event_json['model']['type']
+          if attr == 'data' and model_type == 'ColumnDataSource'
```

Comment:
This pretty much makes `ColumnDataSource` final, though I'm not sure if we can do better with this code anyway. Unless we don't want support this encoding in its subclasses, which may not happen, but at least we have to annotate classes appropriately).